### ISSUE TITLE
Part 10c Environment variables: config difficulties [Windows] 

### ISSUE THREAD

**Issue by tveijola** [fullstack-hy2020/fullstack-hy2020.github.io#767]
Had great difficulties getting the app.json / app.config.js configurations to work on Windows. Ultimately I found a workaround that uses both files like the following:

**app.json**

```
{
  "expo": {
    "name": "rate-repository-app",
    "extra": {},
    "slug": "rate-repository-app",
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#ffffff"
    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": [
      "**/*"
    ],
    "ios": {
      "supportsTablet": true
    },
    "web": {
      "favicon": "./assets/favicon.png"
    }
  }
}
```

**app.config.js**

```
import 'dotenv/config';

const APOLLO_URI = process.env.APOLLO_URI;

export default ({ config }) => ({
  ...config,
  extra: {
    APOLLO_URI
  }
});
```

Note especially the empty "extra" field in `app.json`.
Additionally VSCode complained that 'process' is not defined in `app.config.js`, but this didn't cause any troubles.

**Comment by Kaltsoon** [comment]
Weird indeed, thanks for pointing this out! Did I understand correctly that this fixes the issue:

1. Add `extra: {}` to `app.json`
2. Create the `app.config.js` similar to the one currently in the material, but extend the `app.json` config

**Comment by l0ve2cr3ate** [comment]
I think this issue has to do with `Constants.manifest.extra` being `undefined`: https://github.com/expo/expo/issues/8090? 
I encountered this issue when using Windows and Android device. iOS device was working fine. I changed the `app.json` and `app.config.js` files according to the workaround above. I did also upgrade` expo-cli` to `version 3.27.13`. After this `Constants.manifest.extra` was no longer `undefined`. I then tried if only using the `app.config.json` file like before would still work, and it does. Now `Constants.manifest.extra` is no longer `undefined`. So I am not sure if upgrading the expo-cli did the trick, or if changing the files helped. Maybe someone else who encounters this issue can check it out?
